django/lego/tests/test_dns.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""Unit tests for the dns module."""
+
+from pathlib import Path
+from unittest.mock import MagicMock, patch
+
+from git import Repo
+from lego.dns import remove_dns_record, write_dns_record
+
+
+@patch.object(Path, "exists", False)
+@patch.object(Path, "write_text")
+@patch.object(Repo, "clone_from")
```

Comment:
In the `discourse-gatekeeper` repo, instead of patching, we actually created a local git repo for the tests which reduces the need for mocking and increases the confidence that what we want to do is working. Do you think that would be a valuable approach here?

Reply:
Do you have a link to that test?

Reply:
Here is where we create the fixtures: https://github.com/canonical/discourse-gatekeeper/blob/d49ebe76aaec9fc64947a13728edbb650e69048d/tests/conftest.py#L27

Reply:
Considering the logic here is much simpler, involving checkout, commit and push only, doesn't that seem a bit overkill to you?

Reply:
I'll leave it up to you to decide, I would recommend the approach because interacting with as close as to the real environment with tests as possible means that the tests tend to be a better predictor of whether the application is working. I have myself and seen others create mocking system which makes the unit tests pass whilst everything fails when you deploy to production which is why we did not use the mock approach for discourse gatekeeper as much as possible.

You already have the fixtures you need in the example I linked and it is quite easy to interact with the fixtures to interact with git so that the tests have the files in them that you expect. Perhaps have a go with a time box and go with the mock approach if it takes you too long?

Reply:
That's a good point. I'll change the tests in a follow up PR